lib/core/src/obf.cpp

```diff
@@ -117,40 +119,33 @@ obfSetDebug( int opt ) {
 
 int
 obfiGetFilename( char *fileName ) {
-    char *envVar = NULL;
+    char* envVar{nullptr};
 
     envVar = getRodsEnvAuthFileName();
-    if ( envVar != NULL && *envVar != '\0' ) {
-        strcpy( fileName, envVar );
-        return 0;
-    }
-
-#ifdef windows_platform
-    if ( ProcessType != CLIENT_PT ) {
-        char *tmpstr1;
-        int t;
-        tmpstr1 = iRODSNtGetServerConfigPath();
-        sprintf( fileName, "%s\\irodsA.txt", tmpstr1 );
+    if (envVar != nullptr && *envVar != '\0') {
+        std::strcpy(fileName, envVar);
         return 0;
     }
 
-    /* for clients */
-    envVar = iRODSNt_gethome();
-#else
-    envVar =  getenv( "HOME" );
-#endif
-    if ( envVar == NULL ) {
+    envVar = std::getenv("HOME");
+    if (envVar == nullptr) {
         return ENVIRONMENT_VAR_HOME_NOT_DEFINED;
     }
-    strncpy( fileName, envVar, MAX_NAME_LEN );
-    strncat( fileName, "/", MAX_NAME_LEN );
-    strncat( fileName, AUTH_FILENAME_DEFAULT, MAX_NAME_LEN );
 
-#ifdef windows_platform
-    if ( envVar != NULL ) {
-        free( envVar );
+    using cstr = std::char_traits<char>;
+
+    const auto env_length_with_null_term{std::strlen(envVar) + 1};
+    constexpr auto auth_file_length{cstr::length(AUTH_FILENAME_DEFAULT)};
```

Comment:
Is this preferred over `std::strlen`?

Reply:
I think it's worth considering, at least for some cases. `std::char_traits<char>::length(...)` is constexpr, unlike `std::strlen(...)`, so we could do some more work at compile time.

Reply:
Seems fine.

I'd recommend changing the name of the type alias from `cstr` to something that indicates it is a traits type (e.g. `char_traits`) or using `std::char_traits` in L138 directly.